icechunk-python/tests/test_timetravel.py

```diff
@@ -30,6 +30,7 @@ def test_timetravel():
     assert air_temp[200, 6] == 54
 
     store.checkout(branch="main")
+    store.set_mode("w-")
```

Comment:
This is important. When checking out a branch, we do not change the current mode.

So if you keep switching between a tag and a branch (say) , you'll need to explicitly switch to a writeable mode before making changes.
```
store = IcechunkStore.open_existing(mode="w");
store.checkout(tag=...) # sets mode="r"
store.checkout(branch="main") # preserves mode="r"
store.set_mode("w-") # needed
# make changes
```